caching.md

```diff
@@ -0,0 +1,56 @@
+## Introduction
+
+Intake provides easy access data sources from remote / cloud storage, however for large files the 
+cost of downloading files everytime data is read can be extremely high. We have developed a 
+download once, read many times caching strategy to store and manage data sources on the 
+local file system.
+
+## How it works
+
+Intake catalogs are composed of data sources, which contain attributes descibing how to 
+load data files. To enable caching, catalog authors can now specify caching parameters 
+that will be used to download data file(s), store them in a configured local directory, 
+and return a their location on local disk. For example:
+
+```
+nyc_taxi:
+    description: NYC Taxi dataset
+    driver: csv
+    cache:
+      - argkey: urlpath
+        regex: 'datashader-data'
+        type: file
+    args:
+      urlpath: 's3://datashader-data/nyc_taxi.csv'
+      storage_options: {'anon': True}
+```
+
+Glob strings can be used for the ``urlpath``, which will be expanded to the individual files 
+listed in the base directory of the path. Each file will be downloaded in parallel and referrenced
+in the metadata. From the user's point of view, this happens transparently, they need know nothing
+about the caching policy, but they will notice that read times after the initial access are 
+much faster.
+
+Downloaded files are stored locally in the cache directory, specified in the configuration 
+file, environment variable, or at runtime. The default location for the cache directory 
+is ``~/.intake/cache``. Inside this directory are subdirectories for each data source found in 
+the catalog file that has been cached. These directories are named by hashing the data 
+source driver, urlpath, and cache regex to avoid collision among data sources and cache 
+specifications. Files are downloaded on the first read and will persist until the 
+cache has been cleared explicitly using convience methods. 
```

Comment:
convenience

Reply:
So there's no easy way to have the cached files put into the current directory?  In most cases I'd rather have the files associated with the notebook or project where I'm working, not dumped into an ever-increasing global cache heap. I can see an example below of using get_cwd() explicitly to put it in the current directory, but that's a lot of cruft to put into every notebook.  Is it possible to have some very simple flag for configuring intake (something `intake.local_cache=True`) to store cached items in a subdirectory like `.intake_cache/` in the current directory?

Reply:
I think this is a good idea.  I'll create an issue and we can discuss there.

Reply:
[Issue created](https://github.com/ContinuumIO/intake/issues/151)